providers/aws/resources/aws.lr

```diff
@@ -3014,6 +3014,8 @@ private aws.ec2.instance @defaults("instanceId region state instanceType archite
   instanceType string
   // Tags on the instance
   tags map[string]string
+  // Instance profile of the instance.
+  iamInstanceProfile() dict
```

Comment:
i think it would be a good idea to try to make this a real resource - even if all we show is this information here, the arn and id like we do now -- 
i say this bc we could easily tie together some stuff if we do so, like later on adding aws.iam.instanceprofiles list resource, which could list all instance profiles and have references to the roles attached to those instance profiles, which gives us some good iam role/policy checking that folks could test around for their ec2 instances

all that said, we can also add the dict in now and update later.. but if you have the time, if you could update this to a real resource (and still just return the arn and id like we do now, we dont need to fetch the full instance profile yet), that would set us up for the future, instead of having to later deprecate this one and add another that has the resource reference